### PR TITLE
Feat / add vitest axe plugin

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
## Description

I was playing with the Vitest Axe plugin and here is the result. After adding validation for the Button component, it found one problem when the button was in a busy state.

I added a separate test with multiple button variants to increase the test coverage. We can repeat this pattern for more components this sprint.

```ts
  test('WCAG 2.1 (AA) compliant', async () => {
    const { container } = renderWithRouter(
      <>
        <Button label="Default button" />
        <Button label="Active button" active />
        <Button label="Disabled button" disabled />
        <Button label="Busy button" busy />
        <Button label="Small button" size="small" />
        <Button label="Large button" size="large" />
        <Button label="Full width button" fullWidth />
        <Button label="With start icon" startIcon={<Icon icon={AccountCircle} />} />
        <Button label="Button link" to="https://jwplayer.com" fullWidth />
      </>,
    );

    expect(await axe(container)).toHaveNoViolations();
  });
```

The error reported after the first run:

```
FAIL  src/components/Button/Button.test.tsx > <Button> > WCAG 2.1 (AA) compliant
Error: expect(received).toHaveNoViolations(expected)

Expected the HTML found at $('button:nth-child(4)') to have no violations:

<button class="_button_f8f296 _default_f8f296 _outlined_f8f296" type="button"><span class="_hidden_f8f296">Busy button</span><div class="_buffer_d122df _centerAbsolute_f8f296 _small_d122df"><div></div><div></div><div></div><div></div></div></button>

Received:

"Buttons must have discernible text (button-name)"

Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI
 ❯ src/components/Button/Button.test.tsx:45:34
     43|     );
     44| 
     45|     expect(await axe(container)).toHaveNoViolations();
       |                                  ^
     46|   });
     47| });
```

I also fixed this violation in this PR 😃

### CODE REVIEW

**Review by langemike — APPROVED** [review]
Nice addition and clean-up of unused button code!

**Review by MelissaDTH — COMMENTED** [review]
I've reviewed your add [vitest-axe plugin](https://github.com/Videodock/ott-web-app/pull/101/commits/dc05b91d1de11a2457a8b10a7734a0ae51791f0d) commit, which looks good and clean! But this PR still has some conflicts and also tries to commit 18 other commits as well, is this correct?

**Comment by langemike** [comment]
Shall we merge this with [feat/sprint-3](https://github.com/Videodock/ott-web-app/tree/feat/sprint-3)?

**Comment by ChristiaanScheermeijer** [comment]
@langemike @MelissaDTH I've rebased this PR. It's not a big difference, but there is an a11y fix in this PR... Perhaps we can add more a11y tests in the a11y branch with this added.

**Review by MelissaDTH — APPROVED** [review]

File: packages/ui-react/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -64,7 +62,7 @@ const Button: React.FC<Props> = ({
   const content = (
     <>
       {startIcon && <div className={styles.startIcon}>{startIcon}</div>}
-      {<span className={classNames(styles.buttonLabel, { [styles.hidden]: busy }) || undefined}>{label}</span>}
+      {<span className={classNames({ [styles.hidden]: busy }) || undefined}>{label}</span>}
       {children}
       {busy && <Spinner className={styles.centerAbsolute} size={'small'} />}
```
Comment (royschut) [inline packages/ui-react/src/components/Button/Button.tsx:67]: Should we add an aria-label to the Spinner to tell the screen reader about the busy state?

**Review by royschut — APPROVED** [review]
Nice addition to Vitest!

**Pull request closed without merging**